algebird-core/src/main/scala/com/twitter/algebird/Semigroup.scala

```diff
@@ -70,7 +80,20 @@ class EitherSemigroup[L, R](implicit semigroupl: Semigroup[L], semigroupr: Semig
   }
 }
 
-object Semigroup extends GeneratedSemigroupImplicits with ProductSemigroups {
+class FromAlgebraSemigroup[T](sg: ASemigroup[T]) extends Semigroup[T] {
+  override def plus(l: T, r: T): T = sg.combine(l, r)
+  override def sumOption(ts: TraversableOnce[T]): Option[T] = sg.combineAllOption(ts)
+}
```

Comment:
It's slightly annoying to have to do this, but if you could provide the same thing from `AdditiveSemigroup` as well it would be really helpful.